src/ppx_deriving_yojson.ml

```diff
@@ -319,6 +319,10 @@ let ser_str_of_record ~quoter ~loc varname labels =
   in
   [%expr let fields = [] in [%e assoc]]
 
+let possibly_poly_typ poly_vars ~polymorphize typ =
+  match poly_vars with
+  | [] -> typ
+  | _ -> Typ.poly poly_vars (polymorphize typ)
```

Comment:
could it be better to change ppxlib instead? Replacing the definition of `Typ.poly` by `match poly_vars with [] -> typ | ...` would make sense i think if it's not allowed by the compiler. It would avoid the issue in other ppxs as well

Reply:
Yes that's a good idea. I'll add something in `Ast_builder` though as we don't recommend using `Ast_helper_lite`.

Reply:
Well, turns out `Ast_builder.Default.ptyp_poly` already does that!